SUPPORTED_PLATFORMS.md

```diff
@@ -0,0 +1,46 @@
+# Supported platforms
+
+|  System | Support type | Supported versions | Notes |
+|---|---|---|---|
+| Linux | Tier 1 | Linux >= 2.6.32 with glibc >= 2.12 | |
+| macOS | Tier 1 | macOS >= 10.15 | Current and previous macOS release |
```

Comment:
We dropped 10.15 from CI, so it's no longer Tier 1.

We should also distinguish Intel and M1.

Reply:
Hmm, I just realized that the way the support matrix is a bit weird/flawed. Should the supported version be

- the minimum version we're testing on CI?
- the minimum version that works, *and* we're testing some version (not necessarily the oldest supported version) on CI?

I haven't checked libuv's CI yet but I reckon it's the latter for them (would be pretty surprised if they had tested on windows 8). I wonder if we should follow the same approach?

Reply:
Don't overthink the formalization; the important thing is information/transparency. We should distinguish versions (macOS 11 is Tier 1, macOS 10.15 is Tier 2).